test/powershell/engine/Api/TypeInference.Tests.ps1

```diff
@@ -942,15 +942,44 @@ Describe "Type inference Tests" -tags "CI" {
         $res.Name | Should -Be System.Exception
     }
 
-    It 'Infers type of $_.Expression in typed catch block' {
-        $memberAst = { try {} catch [System.IO.FileNotFoundException] { $_.Exception } }.Ast.Find(
+    $catchClauseTypes = @(
+        @{ Type = 'System.ArgumentException' }
+        @{ Type = 'System.ArgumentNullException' }
+        @{ Type = 'System.ArgumentOutOfRangeException' }
+        @{ Type = 'System.Collections.Generic.KeyNotFoundException' }
+        @{ Type = 'System.DivideByZeroException' }
+        @{ Type = 'System.FormatException' }
+        @{ Type = 'System.IndexOutOfRangeException' }
+        @{ Type = 'System.InvalidOperationException' }
+        @{ Type = 'System.IO.DirectoryNotFoundException' }
+        @{ Type = 'System.IO.DriveNotFoundException' }
+        @{ Type = 'System.IO.FileNotFoundException' }
+        @{ Type = 'System.IO.PathTooLongException' }
+        @{ Type = 'System.Management.Automation.CommandNotFoundException' }
+        @{ Type = 'System.Management.Automation.JobFailedException' }
+        @{ Type = 'System.Management.Automation.RuntimeException' }
+        @{ Type = 'System.Management.Automation.ValidationMetadataException' }
+        @{ Type = 'System.NotImplementedException' }
+        @{ Type = 'System.NotSupportedException' }
+        @{ Type = 'System.ObjectDisposedException' }
+        @{ Type = 'System.OverflowException' }
+        @{ Type = 'System.PlatformNotSupportedException' }
+        @{ Type = 'System.RankException' }
+        @{ Type = 'System.TimeoutException' }
+        @{ Type = 'System.UriFormatException' }
+    )
+
+    It 'Infers type of $_.Expression in [<Type>] typed catch block' -TestCases $catchClauseTypes {
+        param([Type] $Type)
+
+        $memberAst = [scriptblock]::Create("try {} catch [$Type] { $_.Exception }").Ast.Find(
```

Comment:
I meant:
```c#
"try {} catch [System.OverflowException] catch [System.TimeoutException] { $_.Exception }"
```

Reply:
See additional below tests. I suppose I only test `catch [type] {$_.Exception} catch {$_.Exception}` but if you want me to go through a couple stacks I'm game. :)